### PR TITLE
GSoC 2023: Add memcached client project

### PR DESCRIPTION
**Pull request by ktoso**
We also remove one of the previously added projects after thinking through it some more.

### CODE REVIEW

**Comment by ktoso** [comment]
@swift-ci please test

**Review by FranzBusch — APPROVED** [review]

File: gsoc2023/index.md [hunk 1]
```diff
@@ -153,3 +120,26 @@ SwiftPM currently supports a handful of hardcoded templates that can act as a st
...
+- Basic proficiency with Swift 
+- Familiarity with SwiftNIO or other event loop systems
+
+**Description**
+
+We would like to create a Swift native implementation of a [Memcached Client](https://github.com/memcached/memcached) using [SwiftNIO](https://github.com/apple/swift-nio) for the networking stack. The goal is to implement the [binary protocol](https://github.com/memcached/memcached/wiki/BinaryProtocolRevamped) of Memcached to send commands and receives responses. To achieve this we need to implement the request encoding and response decoding. Furthermore, our client should support request pipelining to improve its performance.
+
+**Expected outcomes/benefits/deliverables**
+
+The goal is to support the fundamental commands and expose as an asynchronous first Swift API.
```
Comment (MaxDesiatov) [inline gsoc2023/index.md:141]: nit

```suggestion
The goal is to support the fundamental commands and expose as an asynchronous-first Swift API.
```

**Comment by ktoso** [comment]
@swift-ci please test

**Comment by ktoso** [comment]
Thanks, need an approval from one of the website folks :-)

**Review by 0xTim — APPROVED** [review]

**Pull request merged**